pkg/cmd/roachtest/operations/backup_restore.go

```diff
@@ -42,119 +41,224 @@ func (cl *backupRestoreCleanup) Cleanup(
+}
 
-		err := res.Scan(&id, &tables, &approxRows, &approxBytes, &downloadJobId)
-		if err != nil {
-			o.Fatal(err)
-		}
+func (b *backupRestoreOperation) validateRestore(
+	ctx context.Context, backup backup, restoreDBName string,
+) {
+	b.operation.Status(fmt.Sprintf("verifying db %s matches %s", backup.dbName, restoreDBName))
+	sourceFingerprints, err := fingerprintutils.FingerprintDatabase(ctx, b.conn, backup.dbName, fingerprintutils.AOST(backup.time), fingerprintutils.Stripped())
+	if err != nil {
+		b.operation.Fatal(err)
+	}
 
-		err = tests.WaitForSucceeded(ctx, conn, downloadJobId, 8760*time.Hour /* 1 year - test specs define their own timeouts */)
-		if err != nil {
-			o.Fatal(err)
-		}
+	// No AOST here; the timestamps are rewritten on restore. But nobody else is touching this database, so that's fine.
+	destFingerprints, err := fingerprintutils.FingerprintDatabase(ctx, b.conn, restoreDBName, fingerprintutils.Stripped())
+	if err != nil {
+		b.operation.Fatal(err)
 	}
-	o.Status(fmt.Sprintf("completed restore in %v", timeutil.Since(startTime)))
 
-	if validate {
-		o.Status(fmt.Sprintf("verifying db %s matches %s", dbName, restoreDBName))
-		sourceFingerprints, err := fingerprintutils.FingerprintDatabase(ctx, conn, dbName, fingerprintutils.AOST(backupTS), fingerprintutils.Stripped())
-		if err != nil {
-			o.Fatal(err)
+	if !reflect.DeepEqual(sourceFingerprints, destFingerprints) {
+		b.operation.Fatalf("backup and restore fingerprints do not match: %v != %v", sourceFingerprints, destFingerprints)
+	}
+}
+
+func (b *backupRestoreOperation) scatter(ctx context.Context, dbName string) {
+	var tableNames []string
+
+	b.operation.Status(fmt.Sprintf("scattering database '%s'", dbName))
+
+	rows, err := b.conn.QueryContext(ctx, fmt.Sprintf("SELECT table_name FROM [SHOW TABLES FROM %s]", dbName))
+	if err != nil {
+		b.operation.Fatal(err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var tableName string
+		if err := rows.Scan(&tableName); err != nil {
+			b.operation.Fatal(err)
 		}
+		tableNames = append(tableNames, tableName)
+	}
+	if rows.Err() != nil {
+		b.operation.Fatal(err)
+	}
 
-		// No AOST here; the timestamps are rewritten on restore. But nobody else is touching this database, so that's fine.
-		destFingerprints, err := fingerprintutils.FingerprintDatabase(ctx, conn, restoreDBName, fingerprintutils.Stripped())
+	for _, table := range tableNames {
+		b.operation.Status(fmt.Sprintf("scattering table '%s.%s'", dbName, table))
+		_, err := b.conn.Exec(fmt.Sprintf("ALTER TABLE %s.%s SCATTER", dbName, table))
 		if err != nil {
-			o.Fatal(err)
+			b.operation.Fatal(err)
 		}
+	}
+}
 
-		if !reflect.DeepEqual(sourceFingerprints, destFingerprints) {
-			o.Fatalf("backup and restore fingerprints do not match: %v != %v", sourceFingerprints, destFingerprints)
-		}
+func (b *backupRestoreOperation) RunOfflineRestore(ctx context.Context, skipValidating bool) {
+	dbName := b.pickDB(ctx)
+	backup := b.backup(ctx, dbName, 0)
```

Comment:
previously this operation would run 24 incrementals. Perhaps add back a couple incrementals?